include/swupd.h

```diff
@@ -104,6 +105,7 @@ struct file {
 
 	char *staging; /* output name used during download & staging */
 	CURL *curl;    /* curl handle if downloading */
+	int fd;        /* file written into during downloading, unset when fd_valid is false */
```

Comment:
On Thu, 2016-04-21 at 14:19 -0700, Patrick McCarty wrote:
> In include/swupd.h:
> 
> > @@ -104,6 +105,7 @@ struct file {
> >  
> >  	char *staging; /* output name used during download & staging */
> >  	CURL *curl;    /* curl handle if downloading */
> > +	int fd;        /* file written into during downloading, unset when fd_valid is false */
> 
> I would prefer to use fd == -1 to indicate an invalid file descriptor,
> with the requisite addition of struct file initializers.

That would indeed be the cleaner solution, but introducing such an
initializer in all the right places IMHO goes beyond the scope of this
patch and should better be done by someone who knows the code.

So in other words, it would be good if someone could do that, but please
don't expect me or Dmitry to do it ;-}